components/ai_chat/core/common/mojom/bookmarks.mojom

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+module ai_chat.mojom;
+
+import "url/mojom/url.mojom";
+
+// Represents a bookmark (guaranteed to be non-folder).
+struct Bookmark {
+  int64 id;
+  string title;
+  url.mojom.Url url;
+};
+
+interface BookmarksPageHandler {
+  // Gets the current list of bookmarks.
+  GetBookmarks() => (array<Bookmark> bookmarks);
```

Comment:
When will we call this? Do we need to observe the list and get notified of changes?